src/software/ai/hl/stp/tactic/shadow_freekicker_tactic_test.cpp

```diff
@@ -9,86 +9,794 @@
+     * whether the robot is  blocking the freekick from left or right
+     */
+    void CreateSceneWithoutEnemyRobot(
+        Point ball_position, ShadowFreekickerTactic::FreekickShadower left_or_right)
+    {
+        // set up world, robot, balls
+        World world = ::Test::TestUtil::createBlankTestingWorld();
+
+        world = ::Test::TestUtil::setBallPosition(world, ball_position,
+                                                  Timestamp::fromSeconds(0));
+        world = ::Test::TestUtil::setBallVelocity(world, Vector(0, 0),
+                                                  Timestamp::fromSeconds(0));
+        Robot friendly_robot(0, Point(0, 0), Vector(0, 0), Angle::zero(),
+                             AngularVelocity::zero(), Timestamp::fromSeconds(0));
+        world.mutableFriendlyTeam().updateRobots({friendly_robot});
+
+        // setup tactic and acquire move action from it
+        ShadowFreekickerTactic tactic = ShadowFreekickerTactic(
+            left_or_right, world.enemyTeam(), world.ball(), world.field(), false);
+        tactic.updateRobot(friendly_robot);
+        auto action_ptr = tactic.getNextAction();
+        ASSERT_TRUE(action_ptr);
+        auto move_action = std::dynamic_pointer_cast<MoveAction>(action_ptr);
+        ASSERT_NE(nullptr, move_action);
+
+        // Calculate distance and angle for comparison
+        ball_to_friendly_robot_distance =
+            (world.ball().position() - move_action->getDestination()).length();
+
+        Line ball_to_goal_line =
+            Line(world.ball().position(), world.field().friendlyGoal());
+
+        middle_line_to_friendly_robot_distance =
+            distance(ball_to_goal_line, move_action->getDestination());
+
+        goal_to_ball_angle =
+            (world.ball().position() - world.field().friendlyGoal()).orientation();
+        goal_to_dest_angle =
+            (move_action->getDestination() - world.field().friendlyGoal()).orientation();
+    }
+
+    World world;
+    /**
+     *       Enemy
+     *                       ball
+     *              X
+     *             X X   - -   O - - - - - - - - -  - - - - - - - - - +-----+
+     *              X            .          |
+     *                             .
+     *                             ^ .      |
+     *                             |   .                               Middle line to
+     * friendly robot distance
+     *                +------------+     .  |                            0.09*1.1
+     *   ball to friendly robot distance    +
+     *                                     + +                        +-----+
+     *                            +         +
+     *                            |         | Friendly robot
+     *                            |         |
+     *                            +         +
+     *
+     *                           0.5*1.05+0.09
+     */
+
+    double ball_to_friendly_robot_distance;
+    double ball_to_friendly_robot_distance_abs_error = 0.05;
```

Comment:
minor thing but since this is a constant i think it should have the 'const' keyword and the name should be capitalized, same thing for the other constants

Reply:
That's really good suggestion, I will do that

Reply:
All constants are capitalized and `const` keyword is added